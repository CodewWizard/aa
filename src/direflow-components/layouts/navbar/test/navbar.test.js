import React from "react";
import renderer from "react-test-renderer";
import Navigation from "../navbar";

describe("Navigation", () => {
  const menus = [
    {
      Key: "dashboard",
      Href: "/dashboard",
      Label: "Dashboard",
      Src: "/path/to/dashboard.svg",
      Childs: [],
    },
    {
      Key: "reports",
      Href: "/reports",
      Label: "Reports",
      Src: "/path/to/reports.svg",
      Childs: [
        {
          Key: "sales-report",
          Href: "/reports/sales",
          Label: "Sales Report",
        },
        {
          Key: "expenses-report",
          Href: "/reports/expenses",
          Label: "Expenses Report",
        },
      ],
    },
  ];

  const props = {
    application: { NavigationIndex: 0 },
    Menus: menus,
    onMenuSelected: jest.fn(),
  };

  it("renders correctly", () => {
    const tree = renderer.create(<Navigation {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
