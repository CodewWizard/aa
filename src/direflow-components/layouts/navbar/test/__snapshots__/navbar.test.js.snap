// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav
  className="border-end bgWhite commonShadow active"
  id="sidebar"
>
  <ul
    className="list-unstyled components"
  >
    <li
      className="dropdown active"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <a
        alt=""
        className="dashboard"
        href="/dashboard"
        legacyBehavior={true}
        name="dashboard"
        passHref={true}
      >
        <img
          alt="Dashboard"
          className="svg align-baseline"
          src="/path/to/dashboard.svg"
        />
        <span>
          Dashboard
        </span>
      </a>
    </li>
    <li
      className="dropdown"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <a
        alt=""
        className="dropdown-toggle"
        href="/reports"
        legacyBehavior={true}
        name="reports"
        passHref={true}
      >
        <img
          alt="Reports"
          className="svg align-baseline"
          src="/path/to/reports.svg"
        />
        <span>
          Reports
        </span>
      </a>
      <ul
        className="collapse list-unstyled menu pagesubmenu"
      >
        <li>
          <a
            alt=""
            href="/reports/sales"
          >
            Sales Report
          </a>
        </li>
        <li>
          <a
            alt=""
            href="/reports/expenses"
          >
            Expenses Report
          </a>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;
